File: 卡码/代码随想录/1/06-二叉树/Go/02-二叉树属性/LeetCode_0100.go
//100. 相同的树
package main

import (
	. "binaryTree/types"
)

func isSameTree(p *TreeNode, q *TreeNode) bool {
	if p==nil&&q!=nil {
		return false
	}else if p!=nil&&q==nil {
		return false
	}else if p==nil&&q==nil {
		return true
	}else if p.Val!=q.Val {
		return false
	}else{
		return isSameTree(p.Left,q.Left)&&isSameTree(p.Right,q.Right)
	}
}

File: 卡码/代码随想录/1/06-二叉树/Js/02-二叉树属性/LeetCode_0100.js
//100. 相同的树
import TreeNode from "../TreeNode.js";

const isSameTree = function(p, q) {
  if(!p&&q) return false;
  else if(p&&!q) return false;
  else if(!p&&!q) return true;
  else if(p.val!=q.val) return false;
  else return isSameTree(p.left,q.left)&&isSameTree(p.right,q.right);
};